cmd/publish.go

```diff
@@ -37,14 +37,17 @@ to a published dataset will be immideately visible to connected peers.
 		},
 	}
 
+	cmd.Flags().BoolVarP(&o.Unpublish, "unpublish", "u", false, "unpublish a dataset")
+
 	return cmd
 }
 
 // PublishOptions encapsulates state for the publish command
 type PublishOptions struct {
 	ioes.IOStreams
 
-	Refs []string
+	Refs      []string
+	Unpublish bool
```

Comment:
I'd recommend keeping this consistent with the other flags, not using the negated form, and calling it "Publish" instead, or maybe "PublishState".